zio-http/src/main/scala/zio/http/Server.scala

```diff
@@ -1,300 +1,86 @@
-
-  /**
-   * Creates a new server listening on the provided port.
-   */
-  def withPort(port: Int): Server[R, E] = Concat(self, Server.Address(new InetSocketAddress(port)))
-
-  /**
-   * Creates a new server with netty's HttpContentDecompressor to decompress
-   * Http requests (@see <a href =
-   * "https://netty.io/4.1/api/io/netty/handler/codec/http/HttpContentDecompressor.html">HttpContentDecompressor</a>).
-   */
-  def withRequestDecompression(enabled: Boolean, strict: Boolean): Server[R, E] =
-    Concat(self, RequestDecompression(enabled, strict))
-
-  /**
-   * Creates a new server with the following ssl options.
-   */
-  def withSsl(sslOptions: ServerSSLOptions): Server[R, E] = Concat(self, Server.Ssl(sslOptions))
-
-  /**
-   * Creates a new server by passing a function that modifies the channel
-   * pipeline. This is generally not required as most of the features are
-   * directly supported, however think of this as an escape hatch for more
-   * advanced configurations that are not yet support by ZIO Http.
-   *
-   * NOTE: This method might be dropped in the future.
-   */
-  def withUnsafeChannelPipeline(unsafePipeline: ChannelPipeline => Unit): Server[R, E] =
-    Concat(self, UnsafeChannelPipeline(unsafePipeline))
-
-  /**
-   * Provides unsafe access to netty's ServerBootstrap. Modifying server
-   * bootstrap is generally not advised unless you know what you are doing.
-   */
-  def withUnsafeServerBootstrap(unsafeServerbootstrap: ServerBootstrap => Unit): Server[R, E] =
-    Concat(self, UnsafeServerBootstrap(unsafeServerbootstrap))
+  def port[R]: URIO[R, Int]
 }
-object Server {
-  val disableFlowControl: UServer    = Server.FlowControl(false)
-  val disableLeakDetection: UServer  = LeakDetection(LeakDetectionLevel.DISABLED)
-  val simpleLeakDetection: UServer   = LeakDetection(LeakDetectionLevel.SIMPLE)
-  val advancedLeakDetection: UServer = LeakDetection(LeakDetectionLevel.ADVANCED)
-  val paranoidLeakDetection: UServer = LeakDetection(LeakDetectionLevel.PARANOID)
-  val disableKeepAlive: UServer      = Server.KeepAlive(false)
-  val consolidateFlush: UServer      = ConsolidateFlush(true)
-  private[zio] val log               = Log.withTags("Server")
 
-  def acceptContinue: UServer = Server.AcceptContinue(true)
-
-  def app[R, E](http: HttpApp[R, E]): Server[R, E] = Server.App(http)
-
-  /**
-   * Creates a server from a http app.
-   */
-  def apply[R, E](http: HttpApp[R, E]): Server[R, E] = Server.App(http)
-
-  def bind(port: Int): UServer = Server.Address(new InetSocketAddress(port))
-
-  def bind(hostname: String, port: Int): UServer = Server.Address(new InetSocketAddress(hostname, port))
-
-  def bind(inetAddress: InetAddress, port: Int): UServer = Server.Address(new InetSocketAddress(inetAddress, port))
-
-  def bind(inetSocketAddress: InetSocketAddress): UServer = Server.Address(inetSocketAddress)
+object Server {
+  def serve[R](httpApp: HttpApp[R, Throwable]): URIO[R with Server, Unit] =
```

Comment:
The success type should be `Nothing` to indicate this method never returns.
```suggestion
  def serve[R](httpApp: HttpApp[R, Throwable]): URIO[R with Server, Nothing] =
```

Reply:
done